src/Transition.js

```diff
@@ -316,26 +309,20 @@ class Transition extends React.Component {
 
   onTransitionEnd(timeout, handler) {
     this.setNextCallback(handler)
+
     const node = this.props.nodeRef
       ? this.props.nodeRef.current
       : ReactDOM.findDOMNode(this)
 
-    const doesNotHaveTimeoutOrListener =
-      timeout == null && !this.props.addEndListener
-    if (!node || doesNotHaveTimeoutOrListener) {
-      setTimeout(this.nextCallback, 0)
-      return
-    }
+    if (node && this.props.addEndListener && timeout == null) {
 
-    if (this.props.addEndListener) {
       const [maybeNode, maybeNextCallback] = this.props.nodeRef
         ? [this.nextCallback]
         : [node, this.nextCallback]
-      this.props.addEndListener(maybeNode, maybeNextCallback)
-    }
 
-    if (timeout != null) {
-      setTimeout(this.nextCallback, timeout)
+      this.props.addEndListener(maybeNode, maybeNextCallback)
```

Comment:
this is not equivalent to the original code. previously, if:

- `node` was defined
- `addEndListener` was defined
- `timeout` was defined

then we would call `addEndListener` _and_ we would call `setTimeout` with `this.nextCallback`.

in this revision, there's no way for both `addEndListener` and `setTimeout` to get called

Reply:
Is this the right behaviour for both of them to call `nextCallback`?

Reply:
i think the idea is that, if there's a timeout and an end listener, that the first one that triggers will "win"